src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -469,4 +473,55 @@ TEST_CASE("request_rate_streaming: test that streaming-specific logic works")
   }
 }
 
+TEST_CASE(
+    "custom_json_data: Check custom json data to ensure that it is processed "
+    "correctly")
+{
+  PerfAnalyzerParameters params{};
+  TestRequestRateManager trrm{params};
+
+  std::shared_ptr<MockModelParser> mmp{
+      std::make_shared<MockModelParser>(false, false)};
+  ModelTensor model_tensor{};
+  model_tensor.datatype_ = "INT32";
+  model_tensor.is_optional_ = false;
+  model_tensor.is_shape_tensor_ = false;
+  model_tensor.name_ = "INPUT0";
+  model_tensor.shape_ = {1};
+  mmp->inputs_ = std::make_shared<ModelTensorMap>();
+  (*mmp->inputs_)[model_tensor.name_] = model_tensor;
+
+  std::shared_ptr<MockDataLoader> mdl{std::make_shared<MockDataLoader>()};
+  const std::string json_str{R"(
+{
+  "data": [
+    {
+      "INPUT0": [2123456789]
+    }
+  ]
+}
+    )"};
+  mdl->ReadDataFromStr(json_str, mmp->Inputs(), mmp->Outputs());
+
+  trrm.parser_ = mmp;
+  trrm.data_loader_ = mdl;
+  trrm.using_json_data_ = true;
+  trrm.execute_ = true;
+  trrm.schedule_.push_back(std::chrono::nanoseconds(1));
+
+  std::shared_ptr<TestRequestRateManager::ThreadStat> thread_stat{
+      std::make_shared<TestRequestRateManager::ThreadStat>()};
+  std::shared_ptr<TestRequestRateManager::ThreadConfig> thread_config{
+      std::make_shared<TestRequestRateManager::ThreadConfig>(0, 0)};
+
+  std::future<void> infer_future{std::async(
+      &TestRequestRateManager::Infer, &trrm, thread_stat, thread_config)};
+
+  early_exit = true;
+  infer_future.get();
+
+  REQUIRE(mdl->recorded_inputs_.size() > 0);
+  CHECK(mdl->recorded_inputs_[0] == 2123456789);
```

Comment:
it looks like this intercepts the data at the dataloader. Were you not able to intercept the data all the way out in the mock client?  That would give me more confidence that this is doing what it is supposed to be.

Reply:
I tried to go that route but the data is really difficult, maybe impossible, to access from there. From over there, it's only accessible via a private data member defined in the external C++ library. I'd likely need to change the C++ library to allow access for testing purposes, and that seemed like too much scope.

Reply:
Yikes, I looked closer and it looks like you are right. I'm surprised the client doesn't have access to the data, but I guess that must be for security reasons?  We could use a mock InferInput that overloaded SetSharedMemory and AppendRaw to accumulate data which could then be captured and observed later.  It doesn't look easy to side-swipe in a MockInferInput though. I think we would have to capture creation of it [here](https://github.com/triton-inference-server/client/blob/d0cc4fcdaa6dacfa46652bd8ca06dd2301f72bc6/src/c%2B%2B/perf_analyzer/client_backend/client_backend.cc#L411). 

That seems like a big scope for this ticket.  At the same time I don't know how confident I am that the LoadManagers are doing what they are supposed to just based on this test.  It is mocking a DataLoader function and then capturing inside of the DataLoader. That seems like a dataloader test (which is probably mostly what this ends up being, but I still need to know that the LoadManager is using the dataloader correctly). Does it even matter that we are calling any of the LoadManager functions in this test?

Reply:
That's a great point. This is mostly testing DataLoader functionality. I don't know what to do to test that the request rate manager is sending inputs from JSON other than what I've done. It is really hard to access the inputs from the mock client backend, which I have no idea why. It just seems like there ought to be a place that sends the request and knows that the inputs are but I guess there isn't.

Reply:
I think given there's no easy way to access the input data from the mock client backend, I propose we change this PR to be more of a TestDataLoader PR. Thoughts?

Reply:
I'm thinking that we should mock out InferInput. I'm hoping to plumb that today.  If I'm able to do that, can you use that to confirm end-to-end?

Reply:
If the mock is written to provide public testing access to the raw data from the mock client backend, then yes, I can confirm that amount of JSON data usage end-to-end.